test/e2e_tests/domain_mapping_e2e_test.go

```diff
@@ -64,16 +64,31 @@ var _ = Describe("Validate DomainMapping functionality", func() {
 	})
 
 	It("Should create DomainMapping with secret when Creating an HTTPS Capp instance", func() {
-		By("Creating an https capp")
-		createdCapp, routeHostname, secretName := utilst.CreateHTTPSCapp(k8sClient)
+		By("Creating an HTTP Capp")
+		createdCapp, routeHostname := utilst.CreateCappWithHTTPHostname(k8sClient)
+
+		By("Making sure the tls secret exists in advance")
+		secretName := utilst.GenerateSecretName()
 		secretObject := mock.CreateSecretObject(secretName)
 		utilst.CreateSecret(k8sClient, secretObject)
+		Eventually(func() bool {
+			return utilst.DoesResourceExist(k8sClient, secretObject)
+		}, testconsts.Timeout, testconsts.Interval).Should(BeTrue(), "Should find a resource.")
+
+		By("Changing Capp to be HTTP")
```

Comment:
I might be wrong but shouldn't this be HTTPS not HTTP?

Reply:
You're right I'll change it